src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -61,7 +61,7 @@ internal WebApplicationBuilder(Assembly? callingAssembly, WebApplicationOptions
 
                 // We need to override the application name since the call to Configure will set it to
                 // be the calling assembly's name.
-                webHostBuilder.UseSetting(WebHostDefaults.ApplicationKey, (callingAssembly ?? Assembly.GetEntryAssembly())?.GetName()?.Name ?? string.Empty);
+                webHostBuilder.UseSetting(WebHostDefaults.ApplicationKey, (Assembly.GetEntryAssembly())?.GetName()?.Name ?? string.Empty);
```

Comment:
Do we want people to be able ot set the assembly name directly from the options for the controllers in libraries scenarios?

Reply:
Yes, this is just the default. The API I added (WebApplicationOptions) allows setting the name. Or you can set it via command line args https://github.com/dotnet/aspnetcore/blob/8b06da35ddbc17280f02c0f1dcfd5fac1a40ea72/src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebApplicationTests.cs#L387.

PS: You just made me realize I should update this test to be a different assembly name, now that the entry assembly is the default!